docs/source/create_dataset.mdx

```diff
@@ -0,0 +1,108 @@
+* [`AudioFolder`] uses the [`~datasets.Audio`] feature to decode an audio file. Audio extensions such as wav and mp3 are supported, and you can check the complete [list](https://github.com/huggingface/datasets/blob/b5672a956d5de864e6f5550e493527d962d6ae55/src/datasets/packaged_modules/audiofolder/audiofolder.py#L39) of supported audio extensions.
+
+The dataset splits are generated from the repository structure, and the label names are automatically inferred from the directory name. 
+
+For example, if your image dataset (it is the same for an audio dataset) is stored like this:
+
+```
+pokemon/train/grass/bulbasaur.png
+pokemon/train/fire/charmander.png
+pokemon/train/water/squirtle.png
+
+pokemon/test/grass/ivysaur.png
+pokemon/test/fire/charmeleon.png
+pokemon/test/water/wartortle.png
+```
+
+Then this is how the folder-based builder generates an example:
+
+<div class="flex justify-center">
+    <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/datasets/folder-based-builder.png"/>
+</div>
+
+Create the image dataset by specifying `imagefolder` in [`load_dataset`]:
+
+```py
+>>> from datasets import ImageFolder
+
+>>> dataset = load_dataset("imagefolder", data_dir="/path/to/pokemon")
+```
+
+An audio dataset is created in the same way, except you specify `audiofolder` in [`load_dataset`] instead:
+
+```py
+>>> from datasets import AudioFolder
+
+>>> dataset = load_dataset("audiofolder", data_dir="/path/to/folder")
+```
+
+Any additional information about your dataset, such as text captions or transcriptions, can be included with a `metadata.csv` file in the folder containing your dataset. The metadata file needs to have a `file_name` column that links the image or audio file to its corresponding metadata:
+
+```
+file_name, text
+bulbasaur.png, There is a plant seed on its back right from the day this Pokémon is born.
+charmander.png, It has a preference for hot things.
+squirtle.png, When it retracts its long neck into its shell, it squirts out water with vigorous force.
+```
+
+To learn more about each of these folder-based builders, check out the and <a href="https://huggingface.co/docs/datasets/image_dataset#imagefolder"><span class="underline decoration-yellow-400 decoration-2 font-semibold">ImageFolder</span></a> or <a href="https://huggingface.co/docs/datasets/audio_dataset#audiofolder"><span class="underline decoration-pink-400 decoration-2 font-semibold">AudioFolder</span></a> guides.
+
+## From local files
+
+You can also create a dataset from local files by specifying the path to the data files. There are two ways you can create a dataset using the `from_` methods:
+
+    * The [`~Dataset.from_generator`] method is the most memory-efficient way to create a dataset from a [generator](https://wiki.python.org/moin/Generators) due to a generators lazy behavior. This is especially useful when you"re working with a really large dataset that may not fit in memory.
```

Comment:
I would keep the work "lazy" for iterable datasets

```suggestion
    * The [`~Dataset.from_generator`] method is the most memory-efficient way to create a dataset from a [generator](https://wiki.python.org/moin/Generators) due to a generators iterative behavior. This is especially useful when you"re working with a really large dataset that may not fit in memory, since the dataset is generated on disk progressively and then memory-mapped.
```